### PR TITLE
test: create FLE2 collection and check badge COMPASS-5850 COMPASS-5851

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- [COMPASS-5850](https://jira.mongodb.org/browse/COMPASS-5850) Verify that creating a collection with FLE2 options works
- [COMPASS-5851](https://jira.mongodb.org/browse/COMPASS-5851) Verify that the collection is shown with an FLE2 badge

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts [hunk 1]
```diff
@@ -137,37 +137,28 @@ export async function setConnectFormState(
...
+    if (state.fleKeyVaultNamespace) {
+      await browser.setValueVisible(
+        Selectors.ConnectionFormInputFLEKeyVaultNamespace,
+        state.fleKeyVaultNamespace
+      );
+    }
+    if (state.fleKey) {
+      await browser.expandAccordion(Selectors.ConnectionFormInputFLELocalKMS);
+      await browser.setValueVisible(
+        Selectors.ConnectionFormInputFLEKey,
```
Comment (addaleax) [inline packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts:151]: Just a side note, but I feel like we should namespace selectors for KMS options based on the KMS name; e.g. this should probably be 

```suggestion
        Selectors.ConnectionFormInputFLELocalKey,
```

and the `data-testid` should be `csfle-kms-local-key`, not `csfle-kms-key`

**Review by addaleax — APPROVED** [review]

File: packages/compass-e2e-tests/tests/in-use-encryption.test.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  before(async function () {
+    if (
+      semver.lt(MONGODB_VERSION, '6.0.0-rc0') ||
+      process.env.MONGODB_USE_ENTERPRISE !== 'yes'
+    ) {
+      return this.skip();
+    }
+
+    initialEnvVars = Object.assign({}, process.env);
+    process.env.COMPASS_CSFLE_SUPPORT = 'true';
```
Comment (mcasimir) [inline packages/compass-e2e-tests/tests/in-use-encryption.test.ts:22]: I'm not sure about this .. I have another comment where we set `COMPASS_CSFLE_SUPPORT` in the connection form tests. We decided not to enable FF tests in CI, maybe we need a way to run them in some instances (we should be able to do that with `evergreen patch`). 

The idea is that we need the CI to test the real state of the code that we would be releasing, so to give us the confidence nothing is broken.

For this PR if you enable the FF right after, I think we can keep it, but bear in mind for next time that normally we don't want to enable FF on CI by default.

File: packages/connection-form/package.json [hunk 3]
```diff
@@ -38,9 +38,9 @@
     "depcheck": "depcheck",
     "check": "npm run lint && npm run depcheck",
     "check-ci": "npm run check",
-    "test": "mocha",
-    "test-electron": "xvfb-maybe electron-mocha --no-sandbox",
-    "test-cov": "nyc -x \"**/*.spec.*\" --reporter=lcov --reporter=text --reporter=html npm run test",
+    "test": "cross-env COMPASS_CSFLE_SUPPORT='true' mocha",
```
Comment (mcasimir) [inline packages/connection-form/package.json:41]: The way we normally work with FF locally is this:

```
export COMPASS_CSFLE_SUPPORT='true'

npm run any-kind-of-test
```

This is intentional, we want the CI to always test what's released not what will be released when the FF is enabled.

**Review by mcasimir — APPROVED** [review]
Nice!

**Comment by alenakhineika** [comment]
evergreen retry

**Pull request merged**